core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -199,6 +199,9 @@ private[spark] class BlockManager(
           logError(s"Failed to connect to external shuffle server, will retry ${MAX_ATTEMPTS - i}"
             + s" more times after waiting $SLEEP_TIME_SECS seconds...", e)
           Thread.sleep(SLEEP_TIME_SECS * 1000)
+        case NonFatal(e) =>
+          throw new SparkException("Unable to register with external shuffle server due to : " +
```

Comment:
Not. I wanted to make the exception for this case better as the one which gets shown in the UI with this PR is not definitive.